llvm/utils/TableGen/CodeGenInstruction.cpp

```diff
@@ -325,7 +325,8 @@ static void ParseConstraint(StringRef CStr, CGIOperandList &Ops, Record *Rec) {
 
   // Only other constraint is "TIED_TO" for now.
   StringRef::size_type pos = CStr.find_first_of('=');
-  if (pos == StringRef::npos)
+  if (pos == StringRef::npos || CStr.find_first_of(" \t", pos) != (pos + 1) ||
+      CStr.find_last_of(" \t", pos) != (pos - 1))
```

Comment:
Is it possible for `pos` to be 0 if the constraint starts with `=`? Would `CStr.find_last_of(" \t", pos)` return `StringRef::npos` in that case? Is that equal to `(pos-1)` since StringRef::npos is the largest unsigned number.

Reply:
Won’t this be caught by a different check finding the two operands of an infix operator? If a constraint starts with =, the LHS is missing.

Reply:
Right @topperc -- for the corner case `=`, the behavior is as you describe, so that `npos` == `pos-1` and the condition on `pos+1` fires instead. In other cases, we'll drop through to the following check on the operands as @davidchisnall mentions.

I've added these and other corner cases to the `ConstraintChecking8.td` test file (consolidated).

Reply:
Sure, that's fine too. Commit coming shortly.

Reply:
Done (pos == 0 check).